### PR TITLE
Added examples how to extract gradients to the documentation.

### PR DESCRIPTION
**Pull request by GabrieleMeoni**
# Description

Summary of changes

* ... 
* ...

## Resolved Issues

- [x] fixes #Issue

## How Has This Been Tested?

- [x] Test A
- [x] Test B

## Related Pull Requests

- #PR

### CODE REVIEW

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
             Took 162.147 ms
     
 
-In our case, torchquad was more than 300 times faster than
+In our case, *torchquad* was more than 300 times faster than
 ``scipy.integrate.nquad`` with a just the simple Simpson’s rule. We will add
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:401]:
```suggestion
``scipy.integrate.nquad``. We will add
```

File: docs/source/tutorial.rst [hunk 2]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
             Took 162.147 ms
     
 
-In our case, torchquad was more than 300 times faster than
+In our case, *torchquad* was more than 300 times faster than
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:400]:
```suggestion
In our case, *torchquad*  with Simpson’s rule was more than 300 times faster than
```

File: docs/source/tutorial.rst [hunk 3]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
 ``scipy.integrate.nquad`` with a just the simple Simpson’s rule. We will add
 more elaborate integration methods over time; however, this tutorial should
 already showcase the advantages of numerical integration on the GPU.
 
 Reasonably, one might prefer Monte Carlo integration methods for a 5-D
-problem. We will add this in the future.
\ No newline at end of file
+problem. We will add this in the future.
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:406]:
```suggestion
problem. We might add this comparison to the tutorial in the future.
```

File: docs/source/tutorial.rst [hunk 4]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
+
+Computing gradients with respect to the integration domain
+--------------------------------------------------------
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:409]:
```suggestion
----------------------------------------------------------
```

File: docs/source/tutorial.rst [hunk 5]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
+
+*torchquad* allows fully automatic differentiation. In this tutorial, we will show how to extract the gradients with respect to the integration domain.
+We selected Trapezoid and MonteCarlo methods to showcase that getting the gradients it is possible for both deterministic and stochastic methods.
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:412]:
```suggestion
We selected the Trapezoid rule and the Monte Carlo method to showcase that getting gradients is possible for both deterministic and stochastic methods.
```

File: docs/source/tutorial.rst [hunk 6]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
+    torch.manual_seed(0)  # We have to seed torch to get reproducible results
+    integrators = [MonteCarlo(), Trapezoid()]   # Define integrators
+
+    for integrator in integrators:
+
+        domain = torch.tensor([[-1.0, 1.0]]) #Integration domains
+        domain.requires_grad = True # It enables the creation of a computational graph for gradient calculation.
+        result = integrator.integrate(
+            test_function, dim=1, N=N, integration_domain=domain
+        ) # We calculate the 1 D integral by using the previously defined test-fuction
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:439]:
```suggestion
        ) # We calculate the 1-D integral by using the previously defined test-fuction
```

File: docs/source/tutorial.rst [hunk 7]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
+
+        result.backward() #Gradients computation
+
+        print("Method:", integrator, "Gradients:", domain.grad)
+
+The code above calculates the integral for a 1-D test_fuction `test_function()` in the [-1,1] domain and prints the gradients with respect to the integration domain.
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:445]:
```suggestion
The code above calculates the integral for a 1-D test-function ``test_function()`` in the [-1,1] domain and prints the gradients with respect to the integration domain.
```

File: docs/source/tutorial.rst [hunk 8]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
+The command `domain.requires_grad = True` enables the creation a computational graph, and it shall be called before calling the `integrate(...)` method.
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:446]:
```suggestion
The command ``domain.requires_grad = True`` enables the creation of a computational graph, and it shall be called before calling the ``integrate(...)`` method.
```

File: docs/source/tutorial.rst [hunk 9]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
+Gradients computation is, then, performed calling `result.backward()`.
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:447]:
```suggestion
Gradients computation is, then, performed calling ``result.backward()``. 
```

File: docs/source/tutorial.rst [hunk 10]
```diff
@@ -397,10 +397,57 @@ The results from using Simpson’s rule in *torchquad* is:
...
+
+The code above calculates the integral for a 1-D test_fuction `test_function()` in the [-1,1] domain and prints the gradients with respect to the integration domain.
+The command `domain.requires_grad = True` enables the creation a computational graph, and it shall be called before calling the `integrate(...)` method.
+Gradients computation is, then, performed calling `result.backward()`. 
+The output of the code is as follows:
+
+.. parsed-literal::
+
+    **Output:** Method: <torchquad.integration.monte_carlo.MonteCarlo object at 0x7f724735b6a0> Gradients: tensor([[-1.9872,  2.0150]])
+        Method: <torchquad.integration.trapezoid.Trapezoid object at 0x7f724735b6d0> Gradients: tensor([[-2.0000,  2.0000]])
```
Comment (htoftevaag) [inline docs/source/tutorial.rst:453]:
```suggestion
            Method: <torchquad.integration.trapezoid.Trapezoid object at 0x7f724735b6d0> Gradients: tensor([[-2.0000,  2.0000]])
```

```
.. parsed-literal::

    **Output:** Method: <torchquad.integration.monte_carlo.MonteCarlo object at 0x7f724735b6a0> Gradients: tensor([[-1.9872,  2.0150]])
            Method: <torchquad.integration.trapezoid.Trapezoid object at 0x7f724735b6d0> Gradients: tensor([[-2.0000,  2.0000]])
```

Comment (htoftevaag) [reply docs/source/tutorial.rst:453]: You want to add an extra tab (makes the output cell nicer)

**Review by htoftevaag — CHANGES_REQUESTED** [review]
Nicely done! See suggestions.

**Review by htoftevaag — APPROVED** [review]
Looking good! I'm happy!
